VideoLocker/src/main/java/org/edx/mobile/http/OauthRefreshTokenAuthenticator.java

```diff
@@ -53,14 +56,30 @@ public OauthRefreshTokenAuthenticator(Context context) {
     public Request authenticate(Route route, final Response response) throws IOException {
         logger.warn(response.toString());
 
-        if (!isTokenExpired(response.peekBody(HttpStatus.OK).string())) {
-            return null;
-        }
-
         final AuthResponse currentAuth = loginPrefs.getCurrentAuth();
         if (null == currentAuth || null == currentAuth.refresh_token) {
             return null;
         }
+        String response_body = response.peekBody(200).string();
+
+        switch (getErrorCode(response_body)) {
+            case TOKEN_EXPIRED_ERROR_MESSAGE:
+                break;
```

Comment:
So you want to perform the authentication even if you receive an unknown (non-null) error code? Note that in the current [implementation][1], the `TOKEN_EXPIRED_ERROR_MESSAGE` case is redundant, since it just breaks and there's no `default` case.

[1] https://github.com/edx/edx-app-android/pull/834/commits/1e47c48ae19b94923bb2837de82e45c0792195c7